source/agora/utils/Test.d

```diff
@@ -589,3 +591,174 @@ unittest
+    // Since the amount is evenly distributed in Genesis,
+    // they all have the same value
+    const ExpectedAmount = first.outputs[0].value;
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with twice as many outputs as inputs
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(16).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 16 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 16);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+    auto remainder = ExpectedAmount.div(16);
+    assert(remainder == Amount(0));
+    assert(equalTx.outputs.all!(val => val.value == ExpectedAmount));
+}
+
+/// Test with remainder
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys = iota(3).map!(_ => KeyPair.random()).array;
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], keys.map!(k => k.address).array);
+    // This transaction has 3 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 3);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+    auto remainder = ExpectedAmount.div(3);
+    assert(remainder != Amount(0));
+
+    // The first output includes the remainder.
+    assert(equalTx.outputs[0].value >= ExpectedAmount);
+    assert(equalTx.outputs[1].value == ExpectedAmount);
+    assert(equalTx.outputs[2].value == ExpectedAmount);
+}
+
+/// Test with one output key
+unittest
+{
+    import std.range;
+
+    KeyPair key = KeyPair.random();
+    // Use Genesis
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const equalTx = first.split([genesisKP], [key.address]);
+    // This transaction has 3 txs
+    assert(equalTx.inputs.length == 8);
+    assert(equalTx.outputs.length == 1);
+
+    Amount ExpectedAmount;
+    first.getSumOutput(ExpectedAmount);
+
+    // The first output includes the remainder.
+    assert(equalTx.outputs[0].value == ExpectedAmount);
+}
+
+/// Test with multi input keys
+unittest
+{
+    import std.range;
+
+    KeyPair[] keys8 = iota(8).map!(_ => KeyPair.random()).array;
+    KeyPair genesisKP = getGenesisKeyPair();
+    const first = GenesisBlock.txs[0];
+    const resTx1 = first.split([genesisKP], keys8.map!(k => k.address).array);
+    // This transaction has 8 txs, hence it's just equality
+    assert(resTx1.inputs.length == 8);
+    assert(resTx1.outputs.length == 8);
+    // Since the amount is evenly distributed in Genesis,
+    // they all have the same value
+    const ExpectedAmount1 = first.outputs[0].value;
+    assert(resTx1.outputs.all!(val => val.value == ExpectedAmount1));
```

Comment:
Since this is essentially the first unittest, you could just merge both. We shouldn't copy-paste tests around or it leads to a lot of duplication / resistance when changing things.